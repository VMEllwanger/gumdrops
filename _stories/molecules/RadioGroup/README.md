## Props

The following props may be passed to configure the Radio Group. All additional props will be applied to each `<RadioButton>`

| name      | type       | required | description                                                                          |
| --------- | ---------- | -------- | ------------------------------------------------------------------------------------ |
| className | `String`   |          | Additional class to be added to the outermost element.                               |
| context   | `String`   |          | Indicate the context of the button. Example `default`, `warning`, `danger` etc.      |
| name      | `String`   | true     | Identifier that will be set on individual radio inputs.                              |
| onBlur    | `Function` |          | Blur handler that will be called when a blur event happens on a `<RadioButton>`.     |
| onChange  | `Function` |          | Change handler that will be called when a change event happens on a `<RadioButton>`. |
| options   | `Array`    | true     | Radio button options list. An array of objects with `label` and `value` properties.  |
| size      | `String`   |          | Indicate the size of the button. One of `xs` or `sm`.                                |
| defaultValue | `String or Boolean or Number`|| Select a value by default from the options                                  |

## Example

```javascript
const options = [
    {
        label: 'Foo',
        value: 'foo'
    },
    {
        label: 'Bar',
        value: 'bar'
    }
];

<RadioGroup size="sm" onChange={handleChange} name="my-radio-group" options={options} />;
```
