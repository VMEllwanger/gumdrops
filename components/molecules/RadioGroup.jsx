import React from 'react';
import RadioButton from '../atoms/RadioButton';
import cx from 'classnames';
import PropTypes from 'prop-types';

const RadioGroup = ({ context, size, options, onChange, onBlur, name, className, defaultValue, ...rest }) => {
    const baseClass = 'gds-form-group__radio';

    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass}>
            {options.map(({ label, value }, i) => (
                <RadioButton
                    className="-m-r-4"
                    key={`option-${i}`}
                    label={label}
                    name={name}
                    onBlur={onBlur}
                    context={context}
                    onChange={onChange}
                    value={value}
                    defaultChecked={defaultValue === value}
                    {...rest}
                />
            ))}
        </div>
    );
};

RadioGroup.propTypes = {
    name: PropTypes.string.isRequired,
    /** xs, sm */
    size: PropTypes.oneOf(['xs', 'sm']),
    context: PropTypes.string,
    className: PropTypes.string,
    onChange: PropTypes.func,
    onBlur: PropTypes.func,
    options: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
                .isRequired,
            label: PropTypes.string.isRequired
        })
    ).isRequired,
    defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
};

export default RadioGroup;
