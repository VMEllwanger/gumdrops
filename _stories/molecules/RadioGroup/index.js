import React from 'react';
import { action } from '@storybook/addon-actions';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';
import RadioGroup from '../../../components/molecules/RadioGroup';

const sizeOptions = {
    xs: 'xs',
    sm: 'sm',
    'No Value': ''
};

const contextOptions = {
    success: 'success',
    warning: 'warning',
    danger: 'danger',
    'No Value': ''
};

const component = () => (
    <RadioGroup
        size={optionalSelect('Size', sizeOptions, '')}
        onChange={action('change')}
        onBlur={action('blur')}
        context={optionalSelect('Context', contextOptions, '')}
        name="my-radio-group"
        options={[{ label: 'Foo', value: 'foo' }, { label: 'Bar', value: 'bar' }]}
    />
);

export default [readme, component];
