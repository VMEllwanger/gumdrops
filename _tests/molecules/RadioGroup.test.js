/* globals shallow */
import React from 'react';
import RadioGroup from '../../components/molecules/RadioGroup';

const defaultProps = {
    options: [
        {
            label: 'Cool',
            value: 'cool'
        }
    ],
    className: 'custom-class',
    context: 'success',
    name: 'radio-buttons',
    onBlur: () => {},
    onChange: () => {},
    size: 'sm'
};

describe('Expect <RadioGroup>', () => {
    it('to render', () => {
        const wrapper = shallow(<RadioGroup {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });
});
