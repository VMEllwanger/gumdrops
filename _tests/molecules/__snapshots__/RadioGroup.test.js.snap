// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <RadioGroup> to render 1`] = `
<div
  className="gds-form-group__radio custom-class gds-form-group__radio--sm"
>
  <RadioButton
    className="-m-r-4"
    context="success"
    defaultChecked={false}
    key="option-0"
    label="Cool"
    name="radio-buttons"
    onBlur={[Function]}
    onChange={[Function]}
    value="cool"
  />
</div>
`;
